content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+
+```js
+let errMsg;
+
+const validateSignUp = (email, password, confirmPassword, alert) => {};
+```
+
+We’re passing `alert` as an argument to the function so that we can have access to it when it is imported into the app component. The same thing goes for the other arguments too.
+
+The `errMsg` variable serves as the container that stores and thus displays the error message from the logic. That is why we declared it with the `let` statement, so it can be reassigned to any other value in the code execution process.
+
+- **The conditional statements**
+
+  - The first condition checks if the email field is empty, if it is, an alert error message is rendered in the user interface due to the execution of `alert.error()`
+
+  The string inside the parenthesis is what will be displayed on the UI, and the browser makes sure that the input field is in focus so that it catches the attention of the user.
+
+  ```js
+  if (!email) {
+    alert.error("Please input your email address");
+    emailInput.focus();
+    errMsg = false;
+  }
+  ```
+
+  - The next condition checks if the user enters any password at all. If there’s no value in the form field, an error pops up in the browser, while the next one checks for the validity of the password’s length.
+
+  ```js
+  else if (password === "") {
+      alert.error("Please provide your password!");
+      passwordInput.focus();
+      errMsg = false;
+  } else if (password.value <= 7) {
+      alert.error("Your password must have 8 characters or greater");
+      passwordInput.focus();
+      errMsg = false;
+  }
+  ```
+
+  - This conditional statement checks if there is a match between the first and second passwords. The logic below employs the use of closures in JavaScript to perform this particular validation.
+
+  ```js
+  else if (typeof password !== "undefined" && typeof confirmPassword !== "undefined") {
+      if (password != confirmPassword) {
+          alert.error("Passwords don't match");
+          confirmPasswordInput.focus();
+          errMsg = false;
+      }
+  }
+  ```
+
+  It runs a conditional execution on both password fields to see/make sure that both of them are not undefined (i.e that, they have string values in them). If this first condition passes, the next one will run, if not the next one wouldn’t.
+
+  - If all the conditions stated, have been checked, the alert window comes up with successful text
+
+### Using the validator
+
+Now that we have the validation script set up, and have gotten a grasp of what is going on in the script, it is time to get it inside the app itself. We’d have a look at the step-by-step process of accomplishing that.
+
+First, we need to have it (the validation script) imported into the App component, alongside the react-alert dependency. For brevity’s sake, I wouldn’t be doing much of an explanation on the React component structure, since it is among the pre-requisites of this article.
+
```

Comment:
```suggestion
First, we need to import the validation script into the App component, alongside the react-alert dependency. For brevity’s sake, I wouldn’t be doing much of an explanation on the React component structure, since it is among the pre-requisites of this article.

```